File: java/placedOrdersStatistics/src/generator/RandomDate.java
package generator;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.time.temporal.ChronoUnit;
import java.util.Random;
import java.util.concurrent.ThreadLocalRandom;

public class RandomDate {
    private static final long minDay = LocalDate.of(2020,9,17).toEpochDay();
    private static final long maxDay = LocalDate.now().toEpochDay();

    public static LocalDate getRandomPlacedOrderDate() {
        long randomDay = ThreadLocalRandom.current().nextLong(minDay, maxDay);
        return LocalDate.ofEpochDay(randomDay);
    }

}

File: scala/src/main/scala/generator/RandomDate.scala
package generator

import java.time.format.DateTimeFormatter
import java.time.{LocalDate, LocalDateTime, ZoneOffset}
import java.util.concurrent.ThreadLocalRandom

object RandomDate {
  val minDay: Long = LocalDate.of(2020, 9, 17).toEpochDay
  val maxDay: Long = LocalDate.now().toEpochDay
  val secondsInADay: Long = 86400

  def getRandomPlacedOrderTime: String = {
    val randomDay = ThreadLocalRandom.current().nextLong(minDay, maxDay)
    val randomTime = ThreadLocalRandom.current().nextLong(0, secondsInADay)
    val randomDateTime = LocalDateTime.ofEpochSecond(randomDay * secondsInADay + randomTime, 0, ZoneOffset.UTC)
    val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
    val formattedDateTime = randomDateTime.format(formatter)
    formattedDateTime
  }

}
